### PR TITLE
Comments actions

### PR DESCRIPTION
**Pull request by RyanGC93**
Add edit, like delete,  comment functionnality

### CODE REVIEW

File: react-app/src/Store/comments.js [hunk 1]
```diff
@@ -51,27 +59,31 @@ export const createComment = (userId, postId, content) => async (dispatch) => {
...
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'Application/json'
+      },
+      body: JSON.stringify({ content })
+    }
+    const res = await fetch(`/api/comments/${commentId}`, options)
+    if (!res.ok) alert('issue')
+    const data = await res.json()
+    console.log(data)
```
Comment (jon-wehner) [inline react-app/src/Store/comments.js:75]: remove console.log statement

File: app/api/comment_routes.py [hunk 2]
```diff
@@ -39,3 +39,23 @@ def new_comment():
...
+    db.session.commit()
+    return 'Comment Deleted'
+
+@comment_routes.route('/<int:id>', methods=['PUT'])
+@login_required
+def edit_comment(id):
+    data = request.get_json() 
+    comment = Comment.query.get(id)
+    comment.content = data['content']
+    # userId = data['userId']
```
Comment (jon-wehner) [inline app/api/comment_routes.py:57]: Does this commented code need to be here?

File: react-app/src/components/Comments/comment.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+export default function CommentContent({ comment }) {
+    const history = useHistory()
+    const dispatch = useDispatch();
+    const [isLiked, setIsLiked] = useState(false);
+    const user = useSelector((state) => state.session.user);
+    const dropdownRef = useRef(null);
+    const [isActive, setIsActive] = useDetectOutsideClick(dropdownRef, false);
+
+    const profileRedirect = () => {
+        console.log(comment)
```
Comment (jon-wehner) [inline react-app/src/components/Comments/comment.js:20]: console.log

File: react-app/src/components/Comments/comment.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      };
+      let activeElement
+
+
+    const handleEnter = async(e,commentId) => {
+        // console.log(commentId)
+        activeElement = document.querySelector(`#comment-${comment.id}`)
+
+        let content = activeElement.firstChild.nodeValue
+        console.log(content)
```
Comment (jon-wehner) [inline react-app/src/components/Comments/comment.js:39]: console log statements

File: react-app/src/components/Comments/comment.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        }
+      }
+      
+      const showDropMenu = (comment, user) => {
+          if (comment.userId !== user.id) return
+        setIsActive(!isActive);
+        activeElement = document.querySelector(`#comment-${comment.id}`)
+    
+        // activeElement.classList.add('active')
+        console.log(activeElement)
```
Comment (jon-wehner) [inline react-app/src/components/Comments/comment.js:54]: comment and console.log

File: react-app/src/components/Comments/comment.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                    onClick={() => showDropMenu(comment,user)}
+                            >
+                            {comment.content}
+                            
+                            {comment.userId === user.id && (
+                                <>
+                                <nav
+                                    ref={dropdownRef}
+                                    className={`menu abs ${isActive ? "active" : "inactive"}`}
+                                    // className={`menu abs`}
```
Comment (jon-wehner) [inline react-app/src/components/Comments/comment.js:90]: does this commented code need to be here?

File: react-app/src/components/Comments/dropdown.css [hunk 7]
```diff
@@ -0,0 +1,99 @@
+
+.menu-container {
+    position: relative;
+    display: flex;
+    /* justify-content: center;
+    align-items: center; */
```
Comment (jon-wehner) [inline react-app/src/components/Comments/dropdown.css:6]: comments

File: react-app/src/components/Comments/dropdown.css [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  .menu-trigger {
+    background: #ffffff;
+    cursor: pointer;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    outline: none;
+    /* padding: 4px 6px; */
+
+    /* box-shadow: 0 1px 3px rgba(0, 0, 0, 0.3); */
```
Comment (jon-wehner) [inline react-app/src/components/Comments/dropdown.css:19]: comments

File: react-app/src/components/Comments/dropdown.css [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    transform: translateY(0);
+
+  }
+
+  /* .comment__content.active nav {
+    opacity: 1;
+    visibility: visible;
+    transform: translateY(0);
+
+  } */
```
Comment (jon-wehner) [inline react-app/src/components/Comments/dropdown.css:72]: does this rule need to be here?

File: react-app/src/components/Comments/index.js [hunk 10]
```diff
@@ -1,29 +1,32 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
```
Comment (jon-wehner) [inline react-app/src/components/Comments/index.js:1]: are all of these react imports being used?

File: react-app/src/components/Comments/dropdown.css [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+  nav.abs{
+    right: 0
+  }
+  
+  .menu {
+    background: #ffffff;
+    border-radius: 8px;
+    position: absolute;
+    top: 60px;
+    /* right: 0; */
```
Comment (jon-wehner) [inline react-app/src/components/Comments/dropdown.css:50]: comment

**Review by jon-wehner — CHANGES_REQUESTED** [review]

File: app/api/comment_routes.py (hunk 2 above)
Comment (RyanGC93) [reply app/api/comment_routes.py:57]: remove

File: react-app/src/components/Comments/index.js (hunk 10 above)
Comment (RyanGC93) [reply react-app/src/components/Comments/index.js:1]: yes

**Review by jon-wehner — APPROVED** [review]

**Pull request merged**